File: JAVA/TwoSum.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;


class Solution {
	static class Node
	{
		public int index;
		public int val;
		public Node(int index, int val)
		{
			this.index = index;
			this.val = val;
		}
	}
	static class ValComparator implements Comparator<Node>
	{

		@Override
		public int compare(Node n1, Node n2) {
			return n1.val - n2.val;
		}
		
	}	

    public int[] twoSum(int[] numbers, int target) {
        // Start typing your Java solution below
        // DO NOT write main() function
    	Node[] nodes = new Node[numbers.length];
    	for(int i = 0; i < numbers.length; i++)
    	{
    		nodes[i] = new Node(i+1, numbers[i]);
    	}
    	
        Arrays.sort(nodes,new ValComparator());
        int start = 0;
        int end = numbers.length-1;
        while(start < end)
        {
        	int sum = nodes[start].val + nodes[end].val;
        	if(sum == target)
        	{
        		int[] ret = {nodes[start].index, nodes[end].index};
        		Arrays.sort(ret);
        		return ret;
        	}
        	else if(sum < target)
        		start++;
        	else
        		end--;
        }
        return null;
    }
}
public class TwoSum {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		// TODO Auto-generated method stub

	}

}

File: C++/TwoSum.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<math.h>
#include<map>
using namespace std;
class Node
{
public:
	int index;
	int val;
	Node(int i, int v)
	{
		index = i;
		val = v;
	}
};
bool compare(Node n1, Node n2)
{
	return n1.val < n2.val;
}
class Solution {
public:
	
    vector<int> twoSum(vector<int> &numbers, int target) {
		vector<Node> nodes;
		for(int i = 0; i < numbers.size(); i++)
		{
			nodes.push_back(Node(i+1, numbers[i]));
		}

		sort(nodes.begin(), nodes.end(),compare);
		
		int i = 0;
		int j = nodes.size()-1;
		vector<int> ret;
		while(i < j)
		{
			int sum = nodes[i].val + nodes[j].val;
			if(sum == target)
			{
				ret.push_back(min(nodes[i].index, nodes[j].index));
				ret.push_back(max(nodes[i].index, nodes[j].index));
				return ret;
			}
			else if(sum < target)
				i++;
			else
				j--;
		}
		return ret;
    }
};
int main()
{
	vector<int> nums;
	nums.push_back(5);
	nums.push_back(75);
	nums.push_back(25);
	Solution sol;
	sol.twoSum(nums, 100);
}